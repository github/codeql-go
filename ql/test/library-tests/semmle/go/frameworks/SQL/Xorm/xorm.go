package xormtest

//go:generate depstubber -vendor xorm.io/xorm Engine,Session
//go:generate depstubber -vendor github.com/go-xorm/xorm Engine,Session

import (
	xorm1 "github.com/go-xorm/xorm"
	xorm2 "xorm.io/xorm"
)

func getUntrustedString() string {
	return "trouble"
}

func main() {
	untrusted := getUntrustedString()

	engine1 := xorm1.Engine{}
	engine1.Query(untrusted)
	engine1.QueryString(untrusted)
	engine1.QueryInterface(untrusted)
	engine1.SQL(untrusted)
	engine1.Where(untrusted)
	engine1.Alias(untrusted)
	engine1.NotIn(untrusted)
	engine1.In(untrusted)
	engine1.Select(untrusted)
	engine1.SetExpr(untrusted, nil)
	engine1.OrderBy(untrusted)
	engine1.Having(untrusted)
	engine1.GroupBy(untrusted)

	engine2 := xorm2.Engine{}
	engine2.Query(untrusted)
	engine2.QueryString(untrusted)
	engine2.QueryInterface(untrusted)
	engine2.SQL(untrusted)
	engine2.Where(untrusted)
	engine2.Alias(untrusted)
	engine2.NotIn(untrusted)
	engine2.In(untrusted)
	engine2.Select(untrusted)
	engine2.SetExpr(untrusted, nil)
	engine2.OrderBy(untrusted)
	engine2.Having(untrusted)
	engine2.GroupBy(untrusted)

	session1 := xorm1.Session{}
	session1.Query(untrusted)
	session1.QueryString(untrusted)
	session1.QueryInterface(untrusted)
	session1.SQL(untrusted)
	session1.Where(untrusted)
	session1.Alias(untrusted)
	session1.NotIn(untrusted)
	session1.In(untrusted)
	session1.Select(untrusted)
	session1.SetExpr(untrusted, nil)
	session1.OrderBy(untrusted)
	session1.Having(untrusted)
	session1.GroupBy(untrusted)
	session1.And(untrusted)
	session1.Or(untrusted)

	session2 := xorm2.Session{}
	session2.Query(untrusted)
	session2.QueryString(untrusted)
	session2.QueryInterface(untrusted)
	session2.SQL(untrusted)
	session2.Where(untrusted)
	session2.Alias(untrusted)
	session2.NotIn(untrusted)
	session2.In(untrusted)
	session2.Select(untrusted)
	session2.SetExpr(untrusted, nil)
	session2.OrderBy(untrusted)
	session2.Having(untrusted)
	session2.GroupBy(untrusted)
	session2.And(untrusted)
	session2.Or(untrusted)
}
