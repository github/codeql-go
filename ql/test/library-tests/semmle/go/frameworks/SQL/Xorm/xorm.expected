| xorm.go:19:16:19:24 | untrusted |
| xorm.go:20:22:20:30 | untrusted |
| xorm.go:21:25:21:33 | untrusted |
| xorm.go:22:14:22:22 | untrusted |
| xorm.go:23:16:23:24 | untrusted |
| xorm.go:24:16:24:24 | untrusted |
| xorm.go:25:16:25:24 | untrusted |
| xorm.go:26:13:26:21 | untrusted |
| xorm.go:27:17:27:25 | untrusted |
| xorm.go:28:18:28:26 | untrusted |
| xorm.go:29:18:29:26 | untrusted |
| xorm.go:30:17:30:25 | untrusted |
| xorm.go:31:18:31:26 | untrusted |
| xorm.go:34:16:34:24 | untrusted |
| xorm.go:35:22:35:30 | untrusted |
| xorm.go:36:25:36:33 | untrusted |
| xorm.go:37:14:37:22 | untrusted |
| xorm.go:38:16:38:24 | untrusted |
| xorm.go:39:16:39:24 | untrusted |
| xorm.go:40:16:40:24 | untrusted |
| xorm.go:41:13:41:21 | untrusted |
| xorm.go:42:17:42:25 | untrusted |
| xorm.go:43:18:43:26 | untrusted |
| xorm.go:44:18:44:26 | untrusted |
| xorm.go:45:17:45:25 | untrusted |
| xorm.go:46:18:46:26 | untrusted |
| xorm.go:49:17:49:25 | untrusted |
| xorm.go:50:23:50:31 | untrusted |
| xorm.go:51:26:51:34 | untrusted |
| xorm.go:52:15:52:23 | untrusted |
| xorm.go:53:17:53:25 | untrusted |
| xorm.go:54:17:54:25 | untrusted |
| xorm.go:55:17:55:25 | untrusted |
| xorm.go:56:14:56:22 | untrusted |
| xorm.go:57:18:57:26 | untrusted |
| xorm.go:58:19:58:27 | untrusted |
| xorm.go:59:19:59:27 | untrusted |
| xorm.go:60:18:60:26 | untrusted |
| xorm.go:61:19:61:27 | untrusted |
| xorm.go:62:15:62:23 | untrusted |
| xorm.go:63:14:63:22 | untrusted |
| xorm.go:66:17:66:25 | untrusted |
| xorm.go:67:23:67:31 | untrusted |
| xorm.go:68:26:68:34 | untrusted |
| xorm.go:69:15:69:23 | untrusted |
| xorm.go:70:17:70:25 | untrusted |
| xorm.go:71:17:71:25 | untrusted |
| xorm.go:72:17:72:25 | untrusted |
| xorm.go:73:14:73:22 | untrusted |
| xorm.go:74:18:74:26 | untrusted |
| xorm.go:75:19:75:27 | untrusted |
| xorm.go:76:19:76:27 | untrusted |
| xorm.go:77:18:77:26 | untrusted |
| xorm.go:78:19:78:27 | untrusted |
| xorm.go:79:15:79:23 | untrusted |
| xorm.go:80:14:80:22 | untrusted |
